### DISCUSSION TITLE
Copy script as scratchblocks code

### DISCUSSION THREAD

**Discussion by bobby-sills** [ScratchAddons/ScratchAddons#5587] [Ideas]
It would be nice if you could right-click on a block, and then copy the scratchblocks code for forum use, instead of manually typing it out yourself. scratchblocks currently doesn't have a 3.0 project to scratchblocks generator, only for 2.0, so you would have to make your own generator. (Nevermind, one does exist)

**Comment by CST1229** [comment]
There does exist one: https://apple502j.github.io/parse-sb3-blocks

**Reply by bobby-sills** [reply]
Oh, I didn't notice it on the official scratchblocks website. If it already exists, then I think making this extension should be a little bit easier

**Comment by ksdavidc** [comment]
I mentioned this here https://github.com/ScratchAddons/ScratchAddons/discussions/4901#discussion-4242477. 

Though parse-sb3 can load projects and generate code, it would be more useful, for example for teaching, if I could just get the scratcblocks directly from the editor. For that matter, it might be nice to go the other way, too, i.e. to paste scratchblock text and have it turned into blocks.
